backend/app/adapter/graphql/resolver/changelog.go

```diff
@@ -0,0 +1,33 @@
+package resolver
+
+import (
+	"time"
+
+	"github.com/short-d/short/app/adapter/graphql/scalar"
+	"github.com/short-d/short/app/entity"
+)
+
+// ChangeLog represents ChangeLog entity and user Last Viewed ChangeLog time
+type ChangeLog struct {
+	changeLog    []Change
+	lastViewedAt time.Time
+}
+
+// Changes retrieves full ChangeLog
+func (c ChangeLog) Changes() []Change {
+	return c.changeLog
+}
+
+// LastViewedAt retrieves lastViewedAt for given user
+func (c ChangeLog) LastViewedAt() *scalar.Time {
```

Comment:
Why do we need to have a pointer? If it is not crucial, we can avoid memory escape
https://medium.com/faun/golang-escape-analysis-reduce-pressure-on-gc-6bde1891d625

Reply:
according to schema it is 
`lastViewedAt: Time`
instead of
`lastViewedAt: Time!`
Therefore the resolver function should return a pointer which can be have nil value.

Reply:
@alldroll This is the limitation of the GraphQL framework we are using. It forces pointer when a value can be optional. I think the memory has to be allocated onto the heap.